### PR TITLE
feat(primitives): vanilla portal

### LINKED ISSUE #654: feat(primitives): scaffold @teseor/primitives + 4 v0.3 primitives
## What

Stand up `@teseor/primitives` — one package with sub-paths for framework adapters (`/react`, `/vue`) following the Floating UI shape. Ship the four primitives needed for v0.3 overlay components.

```
packages/primitives/
├── package.json          # exports: ".", "./react", "./vue"
├── src/
│   ├── index.ts          # vanilla barrel
│   ├── focus-trap/
│   │   ├── index.ts
│   │   ├── react.ts
│   │   └── vue.ts
│   ├── portal/           # follow-up
│   ├── dismissable-layer/# follow-up
│   ├── positioning/      # follow-up
│   ├── react.ts          # framework barrel
│   └── vue.ts            # framework barrel
```

## Why

Tooltip and Popover (the v0.3 first overlays per `docs/roadmap.md`) need focus-trap, portal, dismissable-layer, and anchor positioning. The roadmap pulled this package forward from v0.4 specifically so the overlays can ship at v0.3.

One package with sub-paths matches the repo's existing one-package-per-framework convention for the wrappers without forcing three new framework packages for primitives, and gives `@teseor/webc` (#584) a vanilla layer to consume on day one.

## Acceptance

This issue is the umbrella. Each box is a separate PR closing this issue.

- [ ] Scaffold + focus-trap (this PR's scope).
- [ ] Portal (vanilla + React + Vue).
- [ ] Dismissable-layer (ESC + click-outside + pointer-down-outside).
- [ ] Anchor positioning (collision-aware, with autoUpdate hook for resize/scroll).

Per primitive: vanilla function, React hook, Vue composable, vitest unit coverage for each layer, JSDoc explaining the contract.

## Out of scope

- Consumer components (Tooltip / Popover) — separate issues.
- Floating-UI-style middleware framework — keep positioning ergonomic but not pluggable in v0.3.

Roadmap pointer: `docs/roadmap.md` § "v0.3 — Atoms + first headless".

### PR DESCRIPTION
**Pull request by letanure**
Closes #654 (portal checkbox).

## What

Adds `createPortal({ target, container })` to `@teseor/primitives` — appends a container to a target node so content can render outside its natural DOM hierarchy. Defaults to a fresh `<div>` inside `document.body`. `target` accepts any append-capable parent (`Element`, `ShadowRoot`, `DocumentFragment`). The default container is created in the target's `ownerDocument` so a target inside an iframe yields a container belonging to that iframe's document. Returns the container plus an idempotent `unmount` cleanup. 10 happy-dom unit cases cover the defaults, custom target and container, ShadowRoot + DocumentFragment targets, cross-document creation, content hosting, unmount removal, double-unmount safety, independent stacked portals, and a clear-error throw when `document.body` is unavailable.

Exported from `@teseor/primitives` (vanilla). No `/react` or `/vue` sub-path export — see "Why" below.

## Why

Tooltip and Popover need to escape parent stacking and overflow contexts, and `@teseor/webc` and plain-DOM consumers don't have a framework `<Teleport>` or `react-dom.createPortal` to lean on. A vanilla helper that just does "append here, give me a cleanup" is the smallest surface that closes that gap.

No React or Vue adapter ships in this PR by design: React's `createPortal` from `react-dom` already covers the entire React side, and Vue's built-in `<Teleport>` is the idiomatic Vue API. Re-exporting either of those from `@teseor/primitives` adds nothing but visual symmetry. The JSDoc on the vanilla function points framework consumers at their built-ins.

## Test plan

- 10 happy-dom unit cases in `portal/index.test.ts`.
- `pnpm lint && pnpm typecheck && pnpm test` green locally; pre-push gates green.

## Out of scope

- Dismissable-layer and anchor positioning — last two boxes on #654, one PR each (dismissable-layer is up in #657).
- React or Vue portal adapters (see "Why").

### CODE REVIEW

File: packages/primitives/src/portal/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Appends a container to a target node so content can render outside its
+// natural DOM hierarchy. Vanilla DOM only — React consumers use
+// `createPortal` from `react-dom`; Vue consumers use the built-in
+// `<Teleport>`. This vanilla layer exists for `@teseor/webc` and plain-DOM
+// consumers (Astro islands, server-rendered pages, etc.).
+
+export type PortalOptions = {
+  /** Where to attach the portal container. Default: `document.body`. */
+  target?: HTMLElement;
```
[the remark this answers is not in the document]
Comment (letanure) [reply packages/primitives/src/portal/index.ts]: Fixed in c5bb871. `target` is now typed `Element | DocumentFragment` so `HTMLElement`, `ShadowRoot`, and bare `DocumentFragment` all work. Two new tests cover the DocumentFragment and ShadowRoot cases — both portals attach correctly via `parentNode`.

File: packages/primitives/src/portal/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+/**
+ * Creates a DOM container and attaches it to `target` (default
+ * `document.body`). Returns the container plus an `unmount` cleanup. If
+ * `container` is provided it is used as-is; otherwise a fresh `<div>` is
+ * created. Calling `unmount` more than once is a no-op.
+ */
+export function createPortal(options: PortalOptions = {}): Portal {
+  const target = options.target ?? document.body;
```
[the remark this answers is not in the document]
Comment (letanure) [reply packages/primitives/src/portal/index.ts]: Fixed in c5bb871. Default target now goes through a `resolveTarget` helper that checks both `typeof document === 'undefined'` (SSR / worker) and `!document.body` (early scripts, detached docs) and throws `createPortal: no \`target\` provided and \`document.body\` is unavailable`. A test removes `document.body` and asserts the throw (with `try/finally` reattaching the body so the test suite stays isolated).

File: packages/primitives/src/portal/index.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Creates a DOM container and attaches it to `target` (default
+ * `document.body`). Returns the container plus an `unmount` cleanup. If
+ * `container` is provided it is used as-is; otherwise a fresh `<div>` is
+ * created. Calling `unmount` more than once is a no-op. Throws if no
+ * explicit target is given and `document.body` is unavailable.
+ */
+export function createPortal(options: PortalOptions = {}): Portal {
+  const target = resolveTarget(options.target);
+  const container = options.container ?? document.createElement("div");
```
[the remark this answers is not in the document]
Comment (letanure) [reply packages/primitives/src/portal/index.ts]: Fixed in 8504c6e. The default container is now created via `target.ownerDocument.createElement('div')`, so a target from an iframe (or any non-default document) yields a container in that same document. When `options.container` is provided, ownerDocument is irrelevant and isn't checked. If a default is needed but the target has no ownerDocument (edge case for detached / synthetic targets), it throws `createPortal: target has no \`ownerDocument\` and no \`container\` was provided`. New test creates a fresh document via `document.implementation.createHTMLDocument()` and asserts `portal.container.ownerDocument` is that document, not the surrounding one.
